docs/page.rst

```diff
@@ -2329,56 +2329,169 @@ This is an overview of homologous methods on the :ref:`Document` and on the :ref
 ====================================== =====================================
 **Document Level**                     **Page Level**
 ====================================== =====================================
-*Document.get_page_fonts(pno)*         :meth:`Page.get_fonts`
-*Document.get_page_images(pno)*        :meth:`Page.get_images`
-*Document.get_page_pixmap(pno, ...)*   :meth:`Page.get_pixmap`
-*Document.get_page_text(pno, ...)*     :meth:`Page.get_text`
-*Document.search_page_for(pno, ...)*   :meth:`Page.search_for`
+:meth:`Document.get_page_fonts`        :meth:`Page.get_fonts`
+:meth:`Document.get_page_images`       :meth:`Page.get_images`
+:meth:`Document.get_page_pixmap`       :meth:`Page.get_pixmap`
+:meth:`Document.get_page_text`         :meth:`Page.get_text`
+:meth:`Document.search_page_for`       :meth:`Page.search_for`
 ====================================== =====================================
 
-The page number "pno" is a 0-based integer `-∞ < pno < page_count`.
+.. note::
+
+   Most document methods (left column) exist for convenience reasons, and are just wrappers for: *Document[pno].<page method>*. So they **load and discard the page** on each execution.
+
+   However, the first two methods work differently. They only need a page's object definition statement - the page itself will **not** be loaded. So e.g. :meth:`Page.get_fonts` is a wrapper the other way round and defined as follows: `page.get_fonts` == `page.parent.get_page_fonts(page.number)`.
+
+
+When calling the :ref:`Document` equivalent methods then the page number is sent through as a parameter, e.g.:
+
+`Document.get_page_images(pno)` or `Document.get_page_text(pno)`
+
+.. tip::
+
+   The page number parameter, ``pno``, is a 0-based integer `-∞ < pno < page_count`.
+
+
+
+
+
+Tables and Related Classes
+------------------------------------
+
+The `TableFinder` class is returned by :meth:`Page.find_tables` and has related classes as follows:
 
 
 .. class:: TableFinder
 
    An object always returned by :meth:`Page.find_tables`. Attributes of interest:
 
-   ... attribute:: tables
+   .. attribute:: tables
 
-      A list of :ref:`Table` objects, each of which represents a table found on the page. Empty list if no table found.
+      A list of :class:`Table` objects, each of which represents a table found on the page. An empty list if no tables are found.
 
-   ... attribute:: page
+   .. attribute:: page
 
       A reference to the :ref:`Page` object.
 
 
 .. class:: Table
 
-   An object representing a table found on the page. Attributes of interest:
+   An object representing a table found on the page.
+
+
+   .. attribute:: page
+
+      A description of the page instance for the table.
+
+      :type: `string`
```

Comment:
:type: :ref:`Page`

Reply:
This `page` attribute is technically speaking a "weakref proxy" of the `Page` object. Python has this construct to allow such "weak" back-references. They behave like the original object (methods, attributes), but will react gracefully (with an exception - not with a Python interpreter crash) if the object has died in the meantime.
Our approach in this way takes care of situations where users might extract and collect tables across multiple pages: this is forbidden! The only way to achieve this kind of thing is collecting the extracted outputs from `to_markdown`, `to_pandas` etc.